rc/filetype/zig.kak

```diff
@@ -0,0 +1,80 @@
+# zig syntax highlighting for kakoune (https://ziglang.org)
+#
+# based off of https://github.com/ziglang/zig.vim/blob/master/syntax/zig.vim
+# as well as https://ziglang.org/documentation/master/#Grammar
+
+hook global BufCreate .*[.]zig %{
+  set-option buffer filetype zig
+}
+
+hook -group zig-highlight global WinSetOption filetype=zig %{
+    add-highlighter window/zig ref zig
+    hook -once -always window WinSetOption filetype=.* %{ remove-highlighter window/zig }
+}
+
+add-highlighter shared/zig regions
+add-highlighter shared/zig/code default-region group
+
+add-highlighter shared/zig/doc_comment      region '///[^/]' '$'       fill documentation
+add-highlighter shared/zig/comment          region '//' '$'            fill comment
+
+# TODO: highlight escape sequences within strings
+add-highlighter shared/zig/string_double    region '"' (?<!\\)(\\\\)*" fill string
+add-highlighter shared/zig/string_single    region "'" (?<!\\)(\\\\)*' fill string
+add-highlighter shared/zig/string_multiline region '\\\\' '$'          fill string
+
+# storage
+add-highlighter shared/zig/code/ regex '\b(?:const|var|extern|packed|export|pub|noalias|inline|noinline|comptime|callconv|volatile|allowzero|align|linksection|threadlocal)\b' 0:keyword
```

Comment:
There are a bunch of attributes here (e.g. `const`) that could be highlighted using the face of the same name. Maybe you could split that list?

Reply:
Hmm, I believe this entire list is attributes in fact. Good catch, I'll fix that.